### ISSUE TITLE
Allow to get string as a byte array/ROS<byte> from the Utf8JsonReader

### ISSUE THREAD

**Issue by ycrumeyrolle** [dotnet/runtime#1563]
The `Utf8JsonReader` allow to read a string with the method `.GetString()`. 
This method will unescape the characters thanks to `JsonReaderHelper.Unescape()`

It is possible to read a byte array representing a string with the couple ValueSequence/ValueSpan, but the characters will not be escaped. 

For performance reason, it would be great to have an option to read a string as its byte array/ReadOnlySpan<byte> representation, but still unescaped, by exposing a `.GetUtf8String()` to the `Utf8JsonReader`, or by exposing the `JsonReaderHelper.Unescape()` method as public.

**Comment by ahsonkhan** [comment]
The next steps here is to design the API and figure out where such an unescaping capability should live. Should we expose it as an instance method on `Utf8JsonReader`? Are there are other usages of such an unescape API which would warrant that we put it on a type like `JavaScriptEncoder` (since that's were the escaping lives).

To me, this is mainly a static method that takes any arbitrary utf-8 text and tries to unescape it and hence should live on a static helper class rather than an instance on the reader (it doesn't rely on any instance/state of the reader). Do we create a new class for this?

Hence, this requires some API design, but there is definite benefit of having such an API.

**Comment by ycrumeyrolle** [comment]
Yes it is very similar at first glance. 
My use case is to consume the unescaped string without allocating the string. 

I have a use case which is a ~25 hardcoded items switch statement where I have to compare the values. The current implementation is to get the string, then compare the length, then compare the value.
This is 80% faster than a chain of `ValueTextEquals()` calls.
For the moment I have made the statement that data to parse are ASCII so no escape required, except on the '+' character which may be escape or not.

Having the option to the the unescaped JSON string would be useful, mainly if the data are in byte instead of char.

**Comment by eiriktsarpalis** [comment]
> My use case is to consume the unescaped string without allocating the string.

Oh. Wouldn't the existing [`ValueSpan`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.valuespan?view=net-5.0) and [`ValueSequence`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.valuesequence?view=net-5.0) properties allow this?

**Comment by ahsonkhan** [comment]
No, those give you the raw slice from within the JSON representing a particular token, as is, and doesn't do any unescaping.
